src/map/utils/trustutils.cpp

```diff
@@ -345,43 +346,128 @@ void LoadTrustStatsAndSkills(CTrustEntity* PTrust)
 {
     JOBTYPE mJob = PTrust->GetMJob();
     JOBTYPE sJob = PTrust->GetSJob();
-    uint8 mLvl   = PTrust->GetMLevel();
-    uint8 sLvl   = PTrust->GetSLevel();
+    uint8   mLvl = PTrust->GetMLevel();
+    uint8   sLvl = PTrust->GetSLevel();
 
-    // TODO: HP/MP should take into account family, job, etc.
+    // Helpers to map HP/MPScale around 100 to 1-7 grades
+    // std::clamp doesn't play nice with uint8, so -> unsigned int
+    auto mapRanges = [](unsigned int inputStart, unsigned int inputEnd, unsigned int outputStart, unsigned int outputEnd, unsigned int inputVal) -> unsigned int
+    {
+        unsigned int inputRange = inputEnd - inputStart;
+        unsigned int outputRange = outputEnd - outputStart;
+
+        unsigned int output = (inputVal - inputStart) * outputRange / inputRange + outputStart;
+
+        return std::clamp(output, outputStart, outputEnd);
+    };
+
+    auto scaleToGrade = [mapRanges](float input) -> unsigned int
+    {
+        unsigned reverseMappedGrade = mapRanges(80U, 140U, 1U, 7U, static_cast<unsigned int>(input * 100U));
+        reverseMappedGrade += 1U; // Boost grades
+        return std::clamp(7U - reverseMappedGrade, 1U, 7U);
+    };
+
+    // HP/MP ========================
+    // This is the same system as used in charutils.cpp, but modified
+    // to use parts from mob_family_system instead of hardcoded player
```

Comment:
Thank you. I almost balked at the removal of the above todo before seeing this comment. I still believe there will be more redoes of the stat calcs in the future, but _mobs themselves_ need a general cleanup too so accuracy would be debatable either way.